.github/workflows/test-build-deploy-main.yml

```diff
@@ -99,6 +105,8 @@ jobs:
 
       - name: Run testing framework
         run: yarn unit-test --passWithNoTests
+        env:
+          REDIS_HOST: redis
```

Comment:
you should be able to get the whole REDIS_URL from kubernetes using: 

- name: REDIS_URL
              valueFrom:
                secretKeyRef:
                  name: nsm-elasticache
                  key: url
                  
  Instead of setting the host and constructing the URL